### PR TITLE
[KB-264] Rewrite GBR

### PR DESCRIPTION
**Pull request by chhoumann**
Decided to give my own attempt at writing this section.
I've used multiple sources to simplify the section as much as I could. Please let me know if there is anything that is unclear.

I've cut off many of the details that involve the actual trees and how they affect the model.

**See my review comments for how I've merged the two sources & incorporate them in the section.**

### CODE REVIEW

File: report_thesis/src/sections/background.tex [hunk 1]
```diff
@@ -227,28 +227,33 @@ \subsubsection{Extra Trees Regressor (ETR)}
...
 $$
-    \hat{f}(\mathbf{x}) = \sum_{b=1}^{B} \lambda \hat{f}_b(\mathbf{x})
+f_{0}(x)=\arg\min_{\gamma}\sum^{N}_{i=1}L(y_{i},\gamma)
 $$
-In the context of regression, gradient boosting aims to minimize the difference between the predicted values and the actual target values by fitting successive trees to the residuals.
-To minimize errors, gradient descent is used to iteratively update model parameters in the direction of the negative gradient of the loss function, thereby following the path of steepest descent~\cite{gradientLossFunction}.
+where $L(y,\hat{y})$ is the chosen loss function, $N$ is the number of samples, $y_{i}$ are the true values, and $\gamma$ is a constant.
+
+Then we start the iterative process of adding models to the ensemble.
+For each iteration $m$, from $1$ to $M$:
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:245]: This part relies heavily on Hastie.

The algorithm is given in Hastie by:
image

File: report_thesis/src/sections/background.tex [hunk 2]
```diff
@@ -227,28 +227,33 @@ \subsubsection{Extra Trees Regressor (ETR)}
...
+
+\begin{enumerate}
+    \item Compute the residuals of the current model. For regression, this could be the squared error loss, $L(y, \hat{y}) = (y - \hat{y})^2$. The residuals $r_{i}^{(m)}$ for each data point $i$ are calculated as $r_{i}^{(m)} = y_{i} - f_{m-1}(x_{i})$, where $f_{m-1}(x_{i})$ is the prediction of the previous model.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:248]: I use the loss function from Hastie:

image

The definition of the residuals has been _slightly_ modified from the algorithm you see in my above comment (Algorithm 10.3, Hastie). I've opted to simplify the generalized loss function by simply using the squared error loss function directly.

File: report_thesis/src/sections/background.tex [hunk 3]
```diff
@@ -227,28 +227,33 @@ \subsubsection{Extra Trees Regressor (ETR)}
...
+    \item Fit a new model $h_{m}(x)$ to the residuals. This model aims to correct the errors of the current ensemble by using the residuals instead of ground truth values. Essentially, $h_{m}(x)$ tries to predict the residuals $r_{i}^{(m)}$.
+    \item Update the ensemble model by adding the predictions of the new model $h_{m}(x)$, multiplied by a learning rate $\eta$. The learning rate $\eta$ controls the contribution of each new model to the ensemble, preventing overfitting by scaling the updates:
+    $$
+    f_{m}(x)=f_{m-1}(x)+\eta h_{m}(x)
+    $$
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:253]: This is where I draw inspiration from Burkov's work. It much simplifies this part of the algorithm, and introduces the learning rate, which is present in modern implementations. To avoid confusion with the Hastie notation, I use $h$ to denote the function.

image

File: report_thesis/src/sections/background.tex [hunk 4]
```diff
@@ -227,28 +227,33 @@ \subsubsection{Extra Trees Regressor (ETR)}
...
-    \hat{f}^{(b)}(\mathbf{x}) = \hat{f}^{(b-1)}(\mathbf{x}) + \lambda \hat{f}_b(\mathbf{x}),
-$$
-where $\hat{f}_b(\mathbf{x})$ is the prediction of the $b$-th tree and $\lambda$ is the learning rate. Residuals are updated as:
-$$
-    r_i^{(b)} = y_i - \hat{f}^{(b-1)}(\mathbf{x}_i),
-$$
-Each tree is then trained on these updated residuals.
-The repetitive process of fitting a weak learner to predict the residuals and using its predictions to update the model gives us the final model:
+Gradient Boosting is a machine learning technique used for various tasks, including regression and classification.
+The fundamental concept involves sequentially adding models to minimize a loss function, where each successive model addresses the errors of the ensemble of preceding models.
```
Comment (Ivikhostrup) [inline report_thesis/src/sections/background.tex:233]: Would it be worthwhile referring back to decision trees somewhere around here? Given the reworked structure of the background section it might make sense to just drive home the point.

Comment (chhoumann) [reply report_thesis/src/sections/background.tex:233]: Update to specifically mention decision trees. I realized that I scarcely mentioned them at all. I couldn't really find a nice way to integrate a ref back to our section about it. Do you think it's fine that I don't explicitly call back to our section about them? I don't think any other section does, as that section has no label.
